internal/groups/http.go

```diff
@@ -0,0 +1,685 @@
+)
+
+func ListMembership(svc Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(listMemberGroupReq)
+		if err := req.validate(); err != nil {
+			return memberPageRes{}, err
+		}
+		mp, err := svc.ListMemberships(ctx, req.token, req.memberID, req.offset, req.limit, req.metadata)
+		if err != nil {
+			return memberPageRes{}, err
+		}
+		return buildGroupsResponse(mp), nil
+	}
+}
+
+func CreateGroupEndpoint(svc Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(createGroupReq)
+		if err := req.validate(); err != nil {
+			return createGroupRes{}, err
+		}
+
+		group := Group{
+			Name:        req.Name,
+			Description: req.Description,
+			ParentID:    req.ParentID,
+			Metadata:    req.Metadata,
+		}
+
+		gp, err := svc.CreateGroup(ctx, req.token, group)
+		if err != nil {
+			return createGroupRes{}, err
+		}
+		return createGroupRes{
+			created:     true,
+			ID:          gp.ID,
+			ParentID:    gp.ParentID,
+			Description: gp.Description,
+			Metadata:    gp.Metadata,
+			Name:        gp.Name,
+		}, nil
+	}
+}
+
+func ListGroupsEndpoint(svc Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(listGroupsReq)
+		if err := req.validate(); err != nil {
+			return groupPageRes{}, err
+		}
+		gp, err := svc.ListGroups(ctx, req.token, req.offset, req.limit, req.metadata)
+		if err != nil {
+			return groupPageRes{}, err
+		}
+		return buildGroupsResponse(gp), nil
+	}
+}
+
+func ListGroupChildrenEndpoint(svc Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(listGroupsReq)
+		if err := req.validate(); err != nil {
+			return groupPageRes{}, err
+		}
+		gp, err := svc.ListChildren(ctx, req.token, req.groupID, req.offset, req.limit, req.metadata)
+		if err != nil {
+			return groupPageRes{}, err
+		}
+		return buildGroupsResponse(gp), nil
+	}
+}
+
+func ListGroupParentsEndpoint(svc Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(listGroupsReq)
+		if err := req.validate(); err != nil {
+			return groupPageRes{}, err
+		}
+		gp, err := svc.ListParents(ctx, req.token, req.groupID, req.offset, req.limit, req.metadata)
+		if err != nil {
+			return groupPageRes{}, err
+		}
+		return buildGroupsResponse(gp), nil
+	}
+}
+
+func DeleteGroupEndpoint(svc Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(groupReq)
+		if err := req.validate(); err != nil {
+			return nil, err
+		}
+		if err := svc.RemoveGroup(ctx, req.token, req.groupID); err != nil {
+			return nil, err
+		}
+		return groupDeleteRes{}, nil
+	}
+}
+
+func AssignMemberToGroup(svc Service) endpoint.Endpoint {
```

Comment:
Please use consistent naming. We usually use the `Endpoint` suffix. If the method name is too long, you can shorten it like `AssignEndpoint` or `AssignMemberEndpoint`, where `member` implies that we're talking about Groups.